### PR TITLE
KB-11 02/25/2021 Wrote 2 dockerfiles, one for the react frontend, and…

### PR DESCRIPTION
**Pull request by bunny0310**
… one for the django backend, wrote a docker-compose file consisting of three services - frontend, backend, database

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      - CHOKIDAR_USEPOLLING=true
+      
+    stdin_open: true # docker run -i
+    tty: true        # docker run -t
+  
+    container_name: 320_frontend
+
+    networks:
+      - '320'
+
```
Comment (jdnvn) [inline docker-compose.yml:26]: Not major, but remove most of the whitespace between lines. Tidy up / keep clean code for good practice.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    depends_on:
+      - backend
+    
+    volumes:
+      - ./Ghost_in_the_Shell/client_folder_react_app/ethisim:/app
+      - /app/node_modules
+    environment:
+      - CHOKIDAR_USEPOLLING=true
+
```
Comment (marcinknara) [inline docker-compose.yml:18]: Not major, but remove a lot of the whitespace between lines. Tidy up / keep code neat for best practices

Comment (bunny0310) [reply docker-compose.yml]: Done!

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      - "POSTGRES_DB=db"
+    ports:
+      - 5432:5432
+    expose:
+      - '5432'
+    restart: always
+    volumes:
+      - postgres-data:/var/lib/postgresql/data
+    networks:
+      - '320'
```
Comment (marcinknara) [inline docker-compose.yml:47]: I think you should look into Networking configurations for Docker. Based on my own readings I'm not sure if this is a good way to set up custom networks, but if you ran this locally and it works I may be wrong. 

Some links:
https://docs.docker.com/compose/networking/
https://docs.docker.com/network/bridge/#use-the-default-bridge-network

Comment (bunny0310) [reply docker-compose.yml:47]: It worked. In older versions of Docker, if someone wanted to link 2 or more services together, they'd have had to use the 'link' keyword but the correct way to link 2 or more containers such that they are on the same network is to create a new network and use 'depends_on' keyword. This works, I have checked haha.

**Review by marcinknara — APPROVED** [review]
Looks good. Well done testing locally and providing some comments.

**Pull request merged**